src/Parser/Hunter/BeastMastery/Modules/Talents/KillerCobra.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+
+import SPELLS from 'common/SPELLS/index';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SpellUsable from 'Parser/Core/Modules/SpellUsable';
+import { formatNumber } from 'common/format';
+import Haste from 'Parser/Core/Modules/Haste';
+import StatisticBox from "Main/StatisticBox";
+import SpellIcon from "common/SpellIcon";
+
+//Setting it to base cooldown of Kill Command - this works as Killer Cobra gives a total reset regardless of remaining CD.
+const COOLDOWN_REDUCTION_MS = 7500;
+
+class KillerCobra extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    spellUsable: SpellUsable,
+    haste: Haste,
+  };
+
+  effectiveKillCommandResets = 0;
+  wastedKillerCobraCobraShots = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTalent(SPELLS.KILLER_COBRA_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
+    const spellId = event.ability.guid;
+    if (!this.combatants.selected.hasBuff(SPELLS.BESTIAL_WRATH.id)) {
+      return;
+    }
+    if (spellId !== SPELLS.COBRA_SHOT.id) {
+      return;
+    }
+    const killCommandIsOnCooldown = this.spellUsable.isOnCooldown(SPELLS.KILL_COMMAND.id);
+    if (killCommandIsOnCooldown) {
+      this.spellUsable.reduceCooldown(SPELLS.KILL_COMMAND.id, COOLDOWN_REDUCTION_MS);
```

Comment:
you can just use `this.spellUsable.endCooldown(SPELLS.KILL_COMMAND.id);` here, no need for the COOLDOWN_REDUCTION_MS stuff.

Reply:
Nice!